libs/execution_context/ebpf_core.h

```diff
@@ -71,7 +73,20 @@ extern "C"
      */
     ebpf_result_t
     ebpf_core_get_protocol_handler_properties(
-        ebpf_operation_id_t operation_id, _Out_ size_t* minimum_request_size, _Out_ size_t* minimum_reply_size);
+        ebpf_operation_id_t operation_id,
+        _Out_ size_t* minimum_request_size,
+        _Out_ size_t* minimum_reply_size,
+        _Out_ bool* async);
+
+    /**
+     * @brief Cancel a async protocol operation that returned EBPF_PENDING from ebpf_core_invoke_protocol_handler.
+     *
+     * @param[in] async_context Async context passed to ebpf_core_invoke_protocol_handler.
+     * @retval true Operation was cancelled.
```

Comment:
```suggestion
     * @retval true Operation was canceled.
```
nit: American spelling.  Yes we Americans are weird :)